### PR TITLE
Thumbnail upload fixes

### LINKED ISSUE #6082: Thumbnail upload modal doesn't appear if same file is re-selected
image
1. Click "Browse".
2. Select an image.
3. When the modal appears, click "Cancel".
4. Select the same image.  The modal doesn't appear.

### LINKED ISSUE #6097: Channel Create:  Sanitize URL-based thumbnail like in Claim Create
<sub>Original:  [#4481 Issues with new Channel Creation page](https://github.com/lbryio/lbry-desktop/issues/4481) </sub>

The thumbnail-uploader in the Channel Creation page allows URL-based thumbnails, but doesn't check if the URL is valid.  We currently do that for Claim thumbnails, so I think the same method can be repeated here.

Not sure if we current crawl Channel claims or not, but a bad thumbnail would lower the score / make it look bad.

### LINKED ISSUE #6098: Channel Create: Thumbnail uploader shouldn't upload immediately
<sub>Original:  [#4481 Issues with new Channel Creation page](https://github.com/lbryio/lbry-desktop/issues/4481) </sub>

The current thumbnail-upload sequence feels weird:  
1. "Browse" --> sends file
2. "Done"

_I always end up sending the wrong file as I double-clicked the wrong one.  Luckily, nothing embarrassing._

If possible, it should be the following, like in Claim Create:
1. "Browse"
2. "Upload" --> sends file
3. "Done"

### PR DESCRIPTION
**Pull request by saltrafael**
## PR Checklist

Please check all that apply to this PR using "x":

- [X] I have checked that this PR is not a duplicate of an existing PR (open, closed or merged)
- [ ] I have checked that this PR does not introduce a breaking change
- [ ] This PR introduces breaking changes and I have provided a detailed explanation below

## PR Type

What kind of change does this PR introduce?

- [X] Bugfix
- [X] Feature
- [X] Code style update (formatting)
- [X] Refactoring (no functional changes)
- [ ] Documentation changes
- [ ] Other - Please describe:

## Fixes

Issue Number: Closes #6082, closes #6098, closes #6097

### CODE REVIEW

**Comment by tzarebczan** [comment]
Thanks for this! 

It looks good for the thumbnail upload, but this broke the URL feature on the channel page. Can't click Done when adding via URL.

**Comment by saltrafael** [comment]
-made it so the modal remembers the current url value, in case of wanting to edit or erase (currently the form is just reset)
-now I think it closes #6097 :

**Comment by tzarebczan** [comment]
@saltrafael - we recently pushed up changes to these files along with playlists - can you take another stab at this with the current changes?

**Comment by saltrafael** [comment]
@tzarebczan Done!

**Comment by tzarebczan** [comment]
Looks like it's failing - I also tried to rebase on master and it was giving me conflicts. 
image

**Comment by saltrafael** [comment]
@tzarebczan try it now, seems to work for me

File: ui/component/channelEdit/view.jsx [hunk 1]
```diff
@@ -22,6 +22,7 @@ import SUPPORTED_LANGUAGES from 'constants/supported_languages';
 import WalletSpendableBalanceHelp from 'component/walletSpendableBalanceHelp';
 import { SIMPLE_SITE } from 'config';
 import { sortLanguageMap } from 'util/default-languages';
+import ThumbnailBrokenImage from './thumbnail-broken.png';
```
Comment (tzarebczan) [inline ui/component/channelEdit/view.jsx:25]: Don't think this needs a broken png uploaded? https://github.com/lbryio/lbry-desktop/search?q=thumbnail-broken

**Comment by tzarebczan** [comment]
Got to take another close look at this and I think something broke with the pending thumbnails - it would used to show a message that say "your image will show in a few minutes" or something like that. Now it has the red x since the image can't be called via the CDN until after it's confirmed. 

It would be nice if it put the image into the placeholders and then kept it around with the state, or it can just leave that old message. But the red X is bad. My profile was an already uploaded image, so that worked. 

image

**Comment by saltrafael** [comment]
@tzarebczan fixed it now!

The error was supposed to be shown when an invalid URL is given:

[](https://postimg.cc/BXjZ7JR6)

Now it shows back the correct delay message after submitting and unconfirmed by the CDN:

[](https://postimg.cc/YjsMWJjw)

**Comment by saltrafael** [comment]
Moved to https://github.com/lbryio/lbry-desktop/pull/6860

**Pull request closed without merging**